src/sortStudents.ts

```diff
@@ -1,16 +1,59 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married?: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function calcAvgGrade(arrOfGrades: number[]): number {
+  return arrOfGrades.reduce((x: number, y:number) => {
+    return x + y;
+  }) / arrOfGrades.length;
+}
+
+export function sortStudents(
+  students: Student[], sortBy: SortType, order: SortOrder,
+): Student[] {
+  const copyOfSt: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      return copyOfSt.sort((a: Student, b: Student) => {
+        return order === 'asc'
+          ? a[sortBy].localeCompare(b[sortBy])
+          : b[sortBy].localeCompare(a[sortBy]);
+      });
+
+    case SortType.Age:
+    case SortType.Married:
+      return copyOfSt.sort((x: Student, y: Student) => {
+        return order === 'asc'
+          ? x[sortBy] - y[sortBy]
+          : y[sortBy] - x[sortBy];
```

Comment:
```suggestion
          ? +x[sortBy] - +y[sortBy]
          : +y[sortBy] - +x[sortBy];
```
How does it work with boolean values?

Reply:
false == 0
true == 1